New contribute pages + other changes (#40)
elixir-europe/infectious-diseases-toolkit: contribute/editorial_board_guide.md

---
title: Editorial board guide
summary: This guide is there to help editors.
---

## All you need to know about this GitHub repository

### General rules

> As an editor, try to work as much as possible on a different branch than the master branch. This can be on the elixir-europe rdm-toolkit repo or your own fork. Open a pull request if you want to merge your changes with the master branch. This way it is possible for other editors to give feedback on your changes. Typos or other small fixes can of course be done immediately on the master branch.

Reviewer: ```suggestion
As an editor, try to work as much as possible on a different branch than the master branch. This can be on the elixir-europe infectious-diseases-toolkit repo or your own fork. Open a pull request if you want to merge your changes with the master branch. This way it is possible for other editors to give feedback on your changes. Typos or other small fixes can of course be done immediately on the master branch.
```

> As an editor, try to work as much as possible on a different branch than the master branch. This can be on the elixir-europe infectious-diseases-toolkit repo or your own fork. Open a pull request if you want to merge your changes with the master branch. This way it is possible for other editors to give feedback on your changes. Typos or other small fixes can of course be done immediately on the master branch.

Reviewer: ```suggestion
As an editor, try to work as much as possible on a different branch than the master branch. This can be on the elixir-europe infectious-diseases-toolkit repo or your own fork. Open a pull request (PR) if you want to merge your changes with the master branch. This way it is possible for other editors to give feedback on your changes. Typos or other small fixes can of course be done immediately on the master branch.
```

First mention in the page

### The google doc way of contributing
This process is sketched below.

{% include image.html file="googledoc_way_flow.svg" alt="Process of contributing via Google docs" click=true %}

### Overview of the file structure in GitHub

> The content of the website is built up using markdown files found in the [root](https://github.com/elixir-europe/infectious-diseases-toolkit) of the repository, spread in subdirectories according to there website section (human_biomolecular_data, human_clinical_and_health_data, contribute, social_and_economic_impact, ...).

Reviewer: ```suggestion
The content of the website is built up using markdown files found in the [root](https://github.com/elixir-europe/infectious-diseases-toolkit) of the repository, spread in subdirectories according to the website section (human_biomolecular_data, human_clinical_and_health_data, contribute, social_and_economic_impact, ...).
```

### Markdown file naming

- Markdown files should be named without capitals and without spaces (replace them with underscores).
- Make sure that the markdown file has a unique name.

> - If the markdown file is named *example.md*, the page will be found at https://www.infectious-diseases-toolkit.org/.

Reviewer: Is the url given correct? Should it be `https://www.infectious-diseases-toolkit.org/example` instead?

Author: you are right!

Author: but best is actually to give an example with url/section/example like i did the other times

## Review pull requests

If contributors make a pull request to make changes, by default the editors that are responsible for files that will changed by the PR will be assigned and notified. All PR should be assigned to one of the editors. Before merging a PR, pages' tags and keywords, and tools and resources' tags should be checked and assigned according to the established tagging system.
  
## Link a pull request to an issue

When you make a pull request resolving an issue, it is possible to link this pull request to that specific issue. This can be easily done by writing in the conversation of the PR: `closes #number_of_issue`, or `fixes #number_of_issue` or even `resolves #number_of_issue`. This is definitely applicable when authors first open an issue announcing a change or requesting a new page, followed up by the pull request. 
For more information about this topic please visit the [GitHub documentation page](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).

## Adding a new event

> Add an event to the landing page by editing the `events.yml` in the `_data` directory in this repository. Use following attributes to define an event:

Reviewer: ```suggestion
Add an event to the landing page by editing the `events.yml` in the `_data` directory in this repository. Use the following attributes to define an event:
```

```yml
- name: Example event
  startDate: 2021-06-23
  startTime: '9:00'
  endDate: 2021-06-24
  endTime: 13:30 CET
  description: Example description
  location: Online

```

Only name and startDate are mandatory attributes.

## Adding a news item

Add a news item to the landing page by editing the `news.yml` in the `_data` directory in this repository. Use following attributes to define a news item:

```yml
- name: News title
  date: 2021-06-23
  linked_pr: 767
  description: A short description
```

All attributes are mandatory.

## Create a new page

### Simple way: using the GitHub interface

> To generate a new page it is sufficient to simply copy the TEMPLATE file in the subdirectory and rename it. To copy a template you have to:

Reviewer: ```suggestion
To generate a new page it is sufficient to simply copy the TEMPLATE file to the desired subdirectory and rename it. To copy a template you have to:
```
Tried to clarify that the template file needs to be copied to the correct subdirectory, and that the original template file to be copied is not found at the subdirectory

{% include callout.html type="note" content="Always make a new branch when making changes to the website, this to prevent little mistakes and to enforce approval from other editors." %}

### Advanced: working on your own feature branch and pushing local changes

Just like with every change you want to make to this repo, it is possible to do this through Git by working on a local copy. For more information on how to do this, please read our [working with Git](working_with_git) page.

## Find your newly added page back on the website

By default your page will not be linked in the sidebar on the website, or on the landing page, but it will exist as an orphan at *https://www.infectious-diseases-toolkit.org/section/markdown_file_name*. In order to prevent that people will not find the page back it is better to link towards it in the sidebar or get linked within an existing page. 

### Linking pages in the sidebar and frontpage

Make sure all pages are accessible from the navigation sidebar. Please, avoid generating sub-pages that are not directly accessible from the navigation sidebar.

> This website supports multiple sidebars, the one in the main sections of the website is for example different from the one in the contribute section. Both of them are defined by `.yaml` files in the *_data/sidebars* directory. Changing these yaml file will immediately impact the sidebars and the frontpage of the website. The sidebar supports multiple levels and each level in the hierarchy can contain a URL to a page within this website or an external URL.

Reviewer: ```suggestion
This website supports multiple sidebars. For example, the one in the main section of the website is different from the one in the contribute section. Both of them are defined by `.yaml` files in the *_data/sidebars* directory. Changing these yaml files will immediately impact the sidebars and the frontpage of the website. The sidebar supports multiple levels and each level in the hierarchy can contain a URL to a page within this website or an external URL.
```
Shortening the sentence for clarity

> The attributes that define the structure are:

Reviewer: ```suggestion
The attributes that define the sidebar structure are:
```

Avoid manual linking to internal pages. If necessary, you can manually link to the pages like this:

If the markdown page is named example_1.md, you can link towards it using:

```md
[Example 1](/section/example_1)
```

{% include callout.html type="important" content="If you change the file name, you'll have to update all of your links." %}

## Adding extra info to the contributors

Do you want that the GitHub picture of a contributor is shown next to their name? Or maybe you want that the name is clickable and links towards the GitHub page of that person? To enable this please add the name and the necessary metadata to the [CONTRIBUTORS.yaml](https://github.com/elixir-europe/infectious-diseases-toolkit/blob/main/_data/CONTRIBUTORS.yaml) file in the *_data* directory like this:

```yaml
Bert Droesbeke:
    git: bedroesb
    email: bedro@psb.ugent.be
    orcid: 0000-0003-0522-5674
    role: editor
    affiliation: VIB-UGent
```
{% include callout.html type="important" content="Make sure that the name in the yaml file is identically the same as the one used in the metadata of the page." %}

## Adding an institution, infrastructure, project or funder

> Institutions, projects, funders and infrastructures are listed in the [affiliations.yml](https://github.com/elixir-europe/infectious-diseases-toolkit/blob/main/_data/affiliations.yaml) file. The info in this file is used on the support page in the about section, but also for the affiliations in tool assembly pages. Make sure you make use of the same name in those assembly pages. The yaml file has following syntax:

Reviewer: ```suggestion
Institutions, projects, funders and infrastructures are listed in the [affiliations.yml](https://github.com/elixir-europe/infectious-diseases-toolkit/blob/main/_data/affiliations.yaml) file. The information in this file is used on the support page in the about section, but also for the affiliations in showcase pages. Make sure you make use of the same name in those showcase pages. The yaml file has following syntax:
```

  type: institution
  url: https://www.psb.ugent.be/
```

- `name`: name
- `image_url`: relative url towards the image
- `pid`: url including the unique identifier towards the page of the association on [ROR](https://ror.org)
- `expose`: true or false, when true this association will be shown in the about section
- `type`: can be any of these values: *institution*, *funder*, *infrastructure* or *project*
- `url`: url towards the homepage of this association

The logos can be added to the [/images/institutions](https://github.com/elixir-europe/infectious-diseases-toolkit/tree/main/images/institutions/), [/images/projects](https://github.com/elixir-europe/infectious-diseases-toolkit/tree/main/images/projects/), [/images/infrastructures](https://github.com/elixir-europe/infectious-diseases-toolkit/tree/main/images/infrastructures/) and [/images/funders](https://github.com/elixir-europe/infectious-diseases-toolkit/tree/main/images/funders/) directory.

{% include callout.html type="important" content="Upload vector images (.svg filetype) of the institute logo for better quality, scaleability and file size, if possible." %}

## Related pages

### Add "Related pages" to a page 

> Infectious Diseases Toolkit pages from other the sections can be displayed as "Related Infectious Diseases Toolkit pages" in a page, grouped by section. 

Reviewer: ```suggestion
Infectious Diseases Toolkit pages from other sections can be displayed as "Related Infectious Diseases Toolkit pages" in a page, grouped by section. 
```

An overview of all Infectious Diseases Toolkit pages and their `page_id` can be found in the [Website overview page](/contribute/website_overview).

```yml

> related_pages: 
>    your_tasks: [page_id1, page_id2]
>    your_domain: [page_id1, page_id2]
>    tool_assembly: [page_id1, page_id2]

Reviewer: ```suggestion
related_pages: 
   pathogen_characterisation: [page_id1, page_id2]
   social_and_economic_impact: [page_id1, page_id2]
   human_biomolecular_data: [page_id1, page_id2]
   human_clinical_and_health_data: [page_id1, page_id2]
   showcase: [page_id1, page_id2]
```
